gammapy/datasets/map.py

```diff
@@ -2750,26 +2797,31 @@ def _compute_flux_spatial(self):
                 return 1
 
             wcs_geom = self.geom.to_wcs_geom(width_min=self.cutout_width).to_image()
-            values = self.model.spatial_model.integrate_geom(wcs_geom)
-
-            if self.psf and self.model.apply_irf["psf"]:
-                values = self.apply_psf(values)
+            values = self._compute_flux_spatial_geom(wcs_geom)
 
             weights = wcs_geom.region_weights(regions=[self.geom.region])
             value = (values.quantity * weights).sum(axis=(1, 2), keepdims=True)
-
         else:
-            value = self.model.spatial_model.integrate_geom(self.geom)
-            if self.psf and self.model.apply_irf["psf"]:
-                value = self.apply_psf(value)
+            value = self._compute_flux_spatial_geom(self.geom)
 
         return value
 
-    def compute_flux_spatial(self):
-        """Compute spatial flux using caching"""
-        if self.parameters_spatial_changed() or not self.use_cache:
-            self._compute_flux_spatial.cache_clear()
-        return self._compute_flux_spatial()
+    def _compute_flux_spatial_geom(self, geom):
+        """Compute spatial flux oversampling geom if necessary"""
+        factor = self._spatial_oversampling_factor
+        if factor > 1:
+            geom = geom.upsample(factor)
+
+        value = self.model.spatial_model.integrate_geom(geom)
+        if self.psf and self.model.apply_irf["psf"]:
+            psf_kernel_map = self.psf._psf_kernel_map.copy()
+            self.psf._psf_kernel_map = self.psf._psf_kernel_map.upsample(factor)
```

Comment:
I didn't try but this would very likely cause a drop in performance. I think we should assume that the initial binning has been chosen such as it samples correctly the PSF.

Reply:
Is the kernel oversampling even needed? I suppose that the main point is to have properly integrated spatial models. If the standard binning is good enough for the PSF, convolving the integrated spatial model will work fine.

Reply:
I think the "correct" solution here is to evaluate both the model and PSF on the oversampled grid and then downsample. I'd be surprised if there was a large performance difference between interpolation and `.upsample()`.The computational effort of interpolation basically only depends on the final grid that is interpolated on to, which is the same in both cases.

@QRemy I agree on the assumption that the initial binning is chosen well, however the drop in performance is not obvious to me, because we oversample for the PSF kernel computation anyway. The latter could possibly be avoided if we work with the adapted oversampled spatial grid...